### ISSUE TITLE
train without pretrained weights

### ISSUE THREAD

**Issue by iverson333** [198808xc/OrganSegC2F#5]
I tried using the network structure provided in training_C3.prototxt without downloading the pretrained weights. However after ~40,000 iterations the loss was still quite high (~0.98) which means the training simply does not work. 

Before, everything was fine when training started from the FCN-8s pre-trained weights. 

Have you ever tried training without the pretrained weights (i.e. start from scratch) ? Is the training problem I encountered associated with initialization of network weights (I saw the init function in DataC.py, it seems in the `setup` function `self.random = True` already sets the initialization in a random manner)? Or anything else? Thank you.

**Comment by 198808xc** [comment]
Hi,

We never trained this network from scratch. Note that even FCN used pre-trained weights on ImageNet.

I don't quite understand why you hope to train from scratch. Given such a small training set, even you get the training process converged, the accuracy may be considerably low.

BTW, the initialization of network weights has nothing to do with DataC.py nor DataF.py. The setting you mentioned is used to randomly shuffling training data.

If you do have some reasons of training from scratch, please reopen this issue and elaborate.

Best,
Lingxi

**Comment by iverson333** [comment]
Thank you for your response. 

I tried that mainly because I'm a bit suspicious about appying transfer learning in this scenario (a lot differences between natural images and med images). I want to check whether using pretrained model from natural images helps improving results or not. One never knows what it can be without a try.

However I'm not familiar with Caffe and it turns out that the training loss basically does not change. (I just remove the downloading pretrained weights part & `solver.net.copy_from(weights)`)  I really appreciate it if you can tell the correct way of training from scratch.

**Comment by 198808xc** [comment]
You used a correct way of training from scratch, but as I said, it will not work.

Our network is based on FCN, which is fine-tuned based on a pre-trained VGGNet on ImageNet.

Note that, even the VGGNet (please read the VGGNet paper) needs to be initialized using the pre-trained weights on AlexNet.

I just suggest you to stop trying the current strategy. It will not work.

**Comment by iverson333** [comment]
Thanks for your suggestion!
